README.md

```diff
@@ -58,35 +78,134 @@ as a file. All jobs should have the following format:
+2. Global schema overrides apply to all columns of a given name across all
+   tables. They can be specified even for columns that do not exist in every
+   table. They are defined in `config/default_settings.yaml`.
+3. Table schema overrides apply only to the columns of a single table. They
+   take precedence over all other overrides. They are defined in
+   `config/table_definitions.yaml`.
+
+> [!WARNING]
+> `NUMERIC` types are implicitly converted to `DECIMAL(10,0)` because as of
+> 2024, all `NUMERIC` columns without a specified precision and scale are
+> actually just integers. If this changes in the future, it's possible that
+> we could begin to silently truncate numbers via this implicit type
+> conversion. As such, stay on top of schema updates from the iasWorld team.
+
+### Constructing jobs
+
+Predicates, filters, and partitions are Spark concepts used to construct
+individual jobs in a batch. They are mostly handled automatically, but you
+may need to change them in rare cases. The list below outlines the role of each
+concept and how to change them if needed:
 
 - **Predicates** are SQL statements used to chunk a table during reads
   against the iasWorld database. The statements define mutually exclusive
   queries that run in parallel (in order to speed up query execution).
-
   Predicates are defined via a file of SQL statements in the `config/`
   directory, then passed to each table job via a file path.
 - **Filters** are logic conditions included in queries to the database. Spark
   uses [predicate pushdown](https://airbyte.com/data-engineering-resources/predicate-pushdown)
   to compose the _predicates_ and _filter_ for each query into a single SQL
   statement. Think of filters as a SQL WHERE clause applied across all the
   predicate chunks specified above.
-
   Filters are constructed automatically from any `min_year`, `max_year`,
   and/or `cur` values passed as part of a job definition. If these values are
   all null, then the entire table is returned.
 - **Partitions** define how the output Parquet files returned from each should
   be broken up. We use Hive partitioning by default, which yields partitions
   with the structure `$TABLE/taxyr=$YEAR/cur=$CUR_VALUE/part-0.parquet`.
-
   Like filters, partitions are determined automatically via any `min_year`,
   `max_year`, and/or `cur` values that are set. If these values are all null,
   then the table is returned as a single file e.g. `$TABLE/part-0.parquet`.
 
+### Files not included
+
+Some necessary setup and credential files are not included in this repository
+for security reasons. Templated versions are included for instructional
+purposes. If you want to use this repository, you will need to populate the
+following:
+
+- `drivers/ojdbc8.jar` - This is the JDBC driver for our Oracle backend and
+  can be found for free on [Oracle's site](https://www.oracle.com/ca-en/database/technologies/appdev/jdbc-downloads.html).
```

Comment:
**[Question, non-blocking]** I'm guessing this is excluded less for security reasons than for dependency-management reasons, is that right? Or are there credentials in the JDBC driver that aren't populated when downloading from Oracle's site?

Reply:
It's actually a licensing thing. Clarified in 51f1a71.